### PR TITLE
Feature lpdev 219 update markup

### PR DESCRIPTION
**Pull request by aushakov**

### CODE REVIEW

File: src/Lykke.Service.PayInvoicePortal.Repositories/AutofacModule.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+        public AutofacModule(IReloadingManager<string> subscriptionConnectionString, ILog log)
+        {
+            _subscriptionConnectionString = subscriptionConnectionString;
+            _log = log;
+        }
+
+        protected override void Load(ContainerBuilder builder)
+        {
+            builder.RegisterInstance<ISubscriberRepository>(
```
Comment (MikhailGerasimov) [inline src/Lykke.Service.PayInvoicePortal.Repositories/AutofacModule.cs]: Register SubscriberRepository as builder.RegisterType< SubscriberRepository  >.As< ISubscriberRepository > and 
AzureTableStorage as a RegisterInstance. It allow you write unit tests and substitute in-memory storage.

File: src/Lykke.Service.PayInvoicePortal.Repositories/SubscriberRepository.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            var partitionKey = SubscriberEntity.GeneratePartitionKey();
+            return await _tableStorage.GetDataAsync(partitionKey);
+        }
+
+        public async Task<ISubscriber> CreateAsync(ISubscriber subscriber)
+        {
+            var partitionKey = SubscriberEntity.GeneratePartitionKey();
+            var rowKey = SubscriberEntity.GenerateRowKey(subscriber.Email);
+
+            var entity = await _tableStorage.GetDataAsync(partitionKey, rowKey);
```
Comment (MikhailGerasimov) [inline src/Lykke.Service.PayInvoicePortal.Repositories/SubscriberRepository.cs]: It is safe and faster to insert right away. I expect that InsertAsync  throws an exception if there is already exists an row with the same key.

File: src/Lykke.Service.PayInvoicePortal.Services/InvoiceService.cs [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+            {
+                query = query.OrderByDescending(o => o.CreatedDate);
+            }
+
+            if (period != Period.AllTime)
+            {
+                DateTime dateFrom = DateTime.Today;
+
+                if (period == Period.CurrentMonth)
+                    dateFrom = DateTime.Now.Date.AddDays(-DateTime.Now.Day);
```
Comment (MikhailGerasimov) [inline src/Lykke.Service.PayInvoicePortal.Services/InvoiceService.cs]: Should all DateTimes be in UTC?

File: src/Lykke.Service.PayInvoicePortal.Repositories/SubscriberRepository.cs [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+            return newEntity;
+        }
+
+        public async Task DeleteAsync(string email)
+        {
+            var partitionKey = SubscriberEntity.GeneratePartitionKey();
+            var rowKey = email;
+
+            var entity = await _tableStorage.GetDataAsync(partitionKey, rowKey);
```
Comment (tarurar) [inline src/Lykke.Service.PayInvoicePortal.Repositories/SubscriberRepository.cs]: Why do you need to get data before deletion?

File: src/Lykke.Service.PayInvoicePortal/Controllers/Api/EmailController.cs [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+            _invoiceDataService = invoiceDataService;
+            _payInternalClient = payInternalClient;
+            _emailPartnerRouterClient = emailPartnerRouterClient;
+            _log = log;
+        }
+
+        [HttpPost]
+        public async Task<IActionResult> SendAsync([FromBody] EmailSendModel model)
+        {
+            InvoiceModel invoice = await _invoiceDataService.GetById(model.InvoiceId);
```
Comment (tarurar) [inline src/Lykke.Service.PayInvoicePortal/Controllers/Api/EmailController.cs]: GetByIdAsync?

File: src/Lykke.Service.PayInvoicePortal.Repositories/AutofacModule.cs (hunk 1 above)
Comment (aushakov) [reply src/Lykke.Service.PayInvoicePortal.Repositories/AutofacModule.cs]: Yes, sure. Thanks for your advice, but in this case I guess it's Okay.

File: src/Lykke.Service.PayInvoicePortal/Controllers/Api/SubscriberController.cs [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+        [HttpPost]
+        public async Task<IActionResult> Subscribe([FromBody] SubscriberModel model)
+        {
+            if (!ModelState.IsValid)
+            {
+                return Json(new SubscriberResultModel("Wrong email format"));
+            }
+
+            Subscriber subscriber = await _subscriberRepository.GetAsync(model.Email);
```
Comment (samodovdi) [inline src/Lykke.Service.PayInvoicePortal/Controllers/Api/SubscriberController.cs]: model.Email.IsValidEmailAndRowKey() can be used to validate Email aswell

**Pull request merged**